packages/zoe/src/zoeService/startInstance.js

```diff
@@ -0,0 +1,225 @@
+          );
+
+          seatHandleToZoeSeatAdmin.init(seatHandle, zoeSeatAdmin);
+
+          const seatData = harden({
+            proposal,
+            initialAllocation,
+            notifier,
+            seatHandle,
+          });
+
+          zoeSeatAdmins.add(zoeSeatAdmin);
+
+          E(handleOfferObjPromiseKit.promise)
+            .handleOffer(invitationHandle, zoeSeatAdmin, seatData)
+            .then(({ offerResultP, exitObj }) => {
+              offerResultPromiseKit.resolve(offerResultP);
+              exitObjPromiseKit.resolve(exitObj);
+            });
+
+          // return the userSeat before the offerHandler is called
+          return userSeat;
+        },
+        makeNoEscrowSeat: (
+          initialAllocation,
+          proposal,
+          exitObj,
+          seatHandle,
+        ) => {
+          const { userSeat, notifier, zoeSeatAdmin } = makeZoeSeatAdminKit(
+            initialAllocation,
+            exitZoeSeatAdmin,
+            hasExited,
+            proposal,
+            zoeInstanceStorageManager.withdrawPayments,
+            exitObj,
+          );
+          zoeSeatAdmins.add(zoeSeatAdmin);
+          seatHandleToZoeSeatAdmin.init(seatHandle, zoeSeatAdmin);
+          return { userSeat, notifier, zoeSeatAdmin };
+        },
+      });
+      return instanceAdmin;
+    };
+
+    const instanceAdmin = makeInstanceAdmin();
+    zoeInstanceStorageManager.initInstanceAdmin(instance, instanceAdmin);
+
+    E(adminNode)
+      .done()
+      .then(
+        completion => instanceAdmin.exitAllSeats(completion),
+        reason => instanceAdmin.failAllSeats(reason),
+      );
+
+    /** @type {ZoeInstanceAdmin} */
+    const zoeInstanceAdminForZcf = Far('zoeInstanceAdminForZcf', {
+      makeInvitation: zoeInstanceStorageManager.makeInvitation,
+      // checks of keyword done on zcf side
+      saveIssuer: zoeInstanceStorageManager.saveIssuer,
+      // A Seat requested by the contract without any payments to escrow
+      makeNoEscrowSeat: instanceAdmin.makeNoEscrowSeat,
+      exitAllSeats: completion => instanceAdmin.exitAllSeats(completion),
+      failAllSeats: reason => instanceAdmin.failAllSeats(reason),
+      makeZoeMint: zoeInstanceStorageManager.makeZoeMint,
+      replaceAllocations: seatHandleAllocations => {
+        seatHandleAllocations.forEach(({ seatHandle, allocation }) => {
+          const zoeSeatAdmin = seatHandleToZoeSeatAdmin.get(seatHandle);
+          zoeSeatAdmin.replaceAllocation(allocation);
+        });
+      },
+      stopAcceptingOffers: () => instanceAdmin.stopAcceptingOffers(),
+    });
+
+    // At this point, the contract will start executing. All must be
+    // ready
+
+    const {
+      creatorFacet = Far('emptyCreatorFacet', {}),
+      publicFacet = Far('emptyPublicFacet', {}),
+      creatorInvitation: creatorInvitationP,
+      handleOfferObj,
+    } = await E(zcfRoot).executeContract(
+      bundle,
+      zoeServicePromise,
```

Comment:
This used to just be `zoeService`, but it is fine to pass the promise instead, since the contractFacet always uses `E(zoeService)`.

Reply:
gotcha. Resolving.